### ISSUE TITLE
plot.keras_training_history() has warning on learning rate

### ISSUE THREAD

**Issue by gsabot** [rstudio/keras#422]
I trained a model while using both callback_reduce_lr_on_plateau() and
callback_early_stopping().  When I later tried to plot the training history,
I received a warning saying:

```
   Warning message:
   In rbind(values, pad_data) :
     number of columns of result is not a multiple of vector length (arg 2)
```

Some traceback info:

```
   > options(warn=2)
   > plot(val)
   Error in rbind(values, pad_data) : 
     (converted from warning) number of columns of result is not a multiple of vector length (arg 2)
   
   Enter a frame number, or 0 to exit   
   
    1: plot(val)
    2: plot.keras_training_history(val)
    3: as.data.frame(x)
    4: as.data.frame.keras_training_history(x)
    5: rbind(values, pad_data)
    6: .signalSimpleWarning("number of columns of result is not a multiple of vector le
    7: withRestarts({
       .Internal(.signalCondition(simpleWarning(msg, call), msg, cal
    8: withOneRestart(expr, restarts[[1]])
    9: doWithOneRestart(return(expr), restart)
   10: dtk.recover
```
   
In plot.keras_training_history(), it does the following:

```
    ...
    pad_data <- list()
    for (metric in x$params$metrics) pad_data[[metric]] <- rep_len(NA, 
        pad)
    values <- rbind(values, pad_data)
```

At that point in the code I see that x$params$metrics has just 4
values, while x$metrics has 5 named components, and that triggers the
rbind() warning:

```
   Browse[1]> x$params$metrics
   [1] "loss"                    "mean_absolute_error"     "val_loss"               
   [4] "val_mean_absolute_error"

   Browse[1]> names(x$metrics)
   [1] "val_mean_absolute_error" "val_loss"                "lr"                     
   [4] "mean_absolute_error"     "loss"                   
```

I think if the goal is for learning rate/lr to be plotted along with
the other metrics, the for() loop might need to change to iterate over
all five when constructing the padding, like this:

`    for (metric in names(x$metrics) pad_data[[metric]] <- rep_len(NA, 
[val.RData.zip](https://github.com/rstudio/keras/files/2064507/val.RData.zip)
`

Alternatively, if it should not be plotting lr, some other fix is
required?

I've attached the val object from above, a keras_training_history, so
you can call plot(val) to demonstrate the issue and try out that fix.

**Comment by jjallaire** [comment]
Thanks for reporting this! Based on your data and proposed fix I was able to make both fixes (filter out the `lr` metric and include the `lr` metric).

Do you have an opinion about whether we should be plotting `lr` or not?

**Comment by gsabot** [comment]
Given the possibly undiscovered latent issues, fixing the warning for now by
filtering the info out in as.data.frame.keras_training_history() sounds right.

Longer term, it seems like the behavior from the plotting function should be
parallel to the other places you mention.  I'm not familiar enough with those other
callers to have a strong opinion on this.  But if you thought that lr was generally
useful to display alongside the various metrics across most of the callers using
as.data.frame.keras_training_history, then at a later point you might revert the
change while modifying the various callers to show the extra info as appropriate, or
alternatively to be robust to its presence.

**Comment by jjallaire** [comment]
Okay, thanks. That's what I'll do for now.
